### DISCUSSION TITLE
How to creat new blocks

### DISCUSSION THREAD

**Discussion by SHAO-QUAN-LIN** [BIPES/BIPES#98] [Q&A] [answered]
Hey,I want to creat new blocks to add some feature in BIPES,and I've created some blockly by using Blockly Developer Tools.
I wonder how should I  add code in BIPES？

**Comment by gastmaier** [comment, marked as answer]
Hi!
[Check the Creating new blocks tutorial](http://bipes.net.br/docs/get-started/create-block.html) in the official documentation. 

Any doubts about the process feel free to ask more questions.

**Reply by SHAO-QUAN-LIN** [reply]
Thank you very much！I've added some new blockly in bipes_offline.
But I have other problem:I want to send Hexadecimal command to bluetooth 2.0 by using Serial port,and I've tried using "ubluetooth.BLE.gatts_write()","serial.send()","UART.write()" function but when Arduino didn't respond.
How should I send Hexadecimal command to bluetooth 2.0 by using Serial port?

**Reply by rafaelaroca** [reply]
@SHAO-CHUAN-LIN 

Are the devices paired?

Can you use the console? Take a look here: https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_uart_repl.py

We used this example to created the BIPES Bluetooth REPL for MicroPython. 

Check other examples on the link above. We tested all of them and they work nicely.

Rafael
